### PR TITLE
Improve exception handling 

### PR DESCRIPTION
**Pull request by TimGraf**
@AlexMaskovyak @gsiegman @mgonzo @atomboulian @arizonatribe @JonoMercier 

Correct the auth response if auth service is not reachable.

### CODE REVIEW

File: attainia_django_extensions/rpc/rpc_decorator.py [hunk 1]
```diff
@@ -42,16 +48,31 @@ def rpc_error_handler(function):
 
     def wrapper(self, *args, **kwargs):
         """ Call wrapped function """
-        status_code = status.HTTP_201_CREATED if function.__name__ == "create" else status.HTTP_200_OK
-        resp = function(self, *args, **kwargs)
+        logger = logging.getLogger(__name__)
+
+        try:
+            status_code = status.HTTP_201_CREATED if function.__name__ == "create" else status.HTTP_200_OK
```
Comment (AlexMaskovyak) [inline attainia_django_extensions/rpc/rpc_decorator.py]: Python ternaries are so ugly.

**Comment by TimGraf** [comment]
Note that @gsiegman found an issue trying to not use an authentication class while testing and it's not working as expected.  I'm looking into this now.

File: attainia_django_extensions/rpc/rpc_drf.py [hunk 2]
```diff
@@ -155,7 +154,7 @@ def _getJwt(self, request):
         try:
             jwt = get_authorization_header(request).decode().split()[1]
         except Exception as ex:
-            raise exceptions.AuthenticationFailed()
```
Comment (TimGraf) [inline attainia_django_extensions/rpc/rpc_drf.py]: This was needed for the case of not having an authorization class defined.

File: attainia_django_extensions/rpc/rpc_view.py [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 from rest_framework.settings import api_settings
 
 from . import rpc_errors
 
 
 class RpcView(object):
+    logger = logging.getLogger(__name__)
     # The following policies may be set  per-view.
-    authentication_classes = api_settings.DEFAULT_AUTHENTICATION_CLASSES
-    permission_classes = api_settings.DEFAULT_PERMISSION_CLASSES
```
Comment (TimGraf) [inline attainia_django_extensions/rpc/rpc_view.py]: I borrowed this from the DRF code.

https://github.com/encode/django-rest-framework/blob/791539acecb801f3830948dea727f459cbd4b122/rest_framework/views.py#L110

However, I had to remove the init to default auth and permission classes.

File: attainia_django_extensions/rpc/rpc_view.py [hunk 4]
```diff
@@ -90,14 +93,14 @@ def perform_authentication(self):
...
+            except Exception as ex:
+                self.logger.warning("Authenticator failed with error %s", getattr(ex, 'message', repr(ex)))
                 return {rpc_errors.ERRORS_KEY: {rpc_errors.NOT_AUTHENTICATED_KEY: rpc_errors.NOT_AUTHENTICATED_VALUE}}
 
-            if user_auth_tuple is not None:
-                self.request.user, self.request.auth = user_auth_tuple
-                return None
+            return None
 
-        return {rpc_errors.ERRORS_KEY: {rpc_errors.NOT_AUTHENTICATED_KEY: rpc_errors.NOT_AUTHENTICATED_VALUE}}
```
Comment (TimGraf) [inline attainia_django_extensions/rpc/rpc_view.py]: For the case of not having an authentication class defined.

**Comment by TimGraf** [comment]
@AlexMaskovyak and @gsiegman updated to correctly handle the case of not using an authentication or permissions class on views.

**Comment by gsiegman** [comment]
LGTM

**Comment by gsiegman** [comment]
I just tested @TimGraf's fix and it seems to be working great for me now. I'm going to go ahead and merge this as the 2nd thumbs up, because I'm blocked otherwise. 😄

**Comment by AlexMaskovyak** [comment]
Huzzah!

**Pull request merged**